tinygrad/renderer/cstyle.py

```diff
@@ -160,6 +160,8 @@ def ssa(u, prefix="t"):
         val = lang.code_for_op[args](strip_parens(r[vin[0]]), *[r[x] for x in vin[1:]])
       elif args == BinaryOps.MAX:
         val = lang.code_for_op[args](*[lang.render_cast([r[x]], dtype) if x.dtype != dtype else r[x] for x in vin])
+      elif args == TernaryOps.MULACC and any(x.dtype == dtypes.float64 for x in vin):
+        val = lang.code_for_op[args](*[f"(double){r[x]}" for x in vin])
```

Comment:
this is the main fix. the const suffix above doesn't fully address the problem since there's nested `mad` calls.

ex:
```
__kernel void r_2_2_2(__global double* data0, const __global half* data1) {
  int lidx0 = get_local_id(0); /* 2 */
  double acc0 = 0.0l;
  double acc1 = 0.0l;
  float2 val0 = vload_half2(0, data1+lidx0*2);
  data0[lidx0*2] = mad((val0).y,0.0l,mad((val0).x,1.0l,acc0));
  data0[(lidx0*2)+1] = mad((val0).y,1.0l,mad((val0).x,0.0l,acc1));
}
```

becomes
```
__kernel void r_2_2_2(__global double* data0, const __global half* data1) {
  int lidx0 = get_local_id(0); /* 2 */
  double acc0 = 0.0l;
  double acc1 = 0.0l;
  float2 val0 = vload_half2(0, data1+lidx0*2);
  data0[lidx0*2] = mad((double)(val0).y,(double)0.0l,(double)mad((double)(val0).x,(double)1.0l,(double)acc0));
  data0[(lidx0*2)+1] = mad((double)(val0).y,(double)1.0l,(double)mad((double)(val0).x,(double)0.0l,(double)acc1));
}
```